pandas/tests/test_reshape.py

```diff
@@ -326,6 +326,20 @@ def test_include_na(self):
                                 dtype=np.uint8)
         tm.assert_numpy_array_equal(res_just_na.values, exp_just_na.values)
 
+    def test_keep_na(self):
+        s = ['a', 'b', np.nan]
+        res_na = get_dummies(s, dummy_na='keep', sparse=self.sparse)
+        exp_na = DataFrame({'a': {0: 1, 1: 0, 2: np.nan},
+                            'b': {0: 0, 1: 1, 2: np.nan}},
+                           dtype=np.float16)
```

Comment:
yeah ``float32`` is the best can do for now. (and FYI that's a nother reason to drop the NaN's, then you *can* use a small int type).